### ISSUE TITLE
content filter not working

### ISSUE THREAD

**Issue by jbarraca** [theonemule/docker-proxy-server#1]
Hi, i tried to instantiate this docker-proxy on a docker container running on a VM and it isnt working.
Instead of using a blacklist i used the sample that is already in the repository.
![image](https://user-images.githubusercontent.com/33351821/36675437-d782a320-1b00-11e8-9650-935b8eb88022.png)
after running the container
![image](https://user-images.githubusercontent.com/33351821/36675493-0e07fd8c-1b01-11e8-9dd2-e69dd4cd3d5e.png)
i added the rules to iptables:
`sudo iptables -t nat -A PREROUTING -p tcp --dport 80 -j DNAT --to-destination 10.203.0.80:3128

sudo iptables -t nat -A PREROUTING -p tcp --dport 443 -j DNAT --to-destination 10.203.0.80:3128

sudo iptables -t nat -A POSTROUTING -j MASQUERADE`

when i try to access the domain on the blacklist (example.com), instead of being redirected to the block.html page, i receive the content of example.com:
![image](https://user-images.githubusercontent.com/33351821/36675624-7224ada6-1b01-11e8-95cc-b84ece386b85.png)

can you help me understand what am i doing wrong?
Thanks.

**Comment by theonemule** [comment]
Could you try it by removing the -p directives in your docker run command. Docker uses IP Tables too, so I think that might be interfering with it the rules you created.

**Comment by jbarraca** [comment]
![image](https://user-images.githubusercontent.com/33351821/36724509-508934e6-1bab-11e8-9783-9489206b8608.png)
It still doesnt work

**Comment by theonemule** [comment]
Did you re-add the iptables rules after you removed the container?

Can you show the output of iptables -t nat -L

**Comment by jbarraca** [comment]
Yes, i did re-add the iptables.

![image](https://user-images.githubusercontent.com/33351821/36785186-e92a6d0e-1c79-11e8-9ee6-40c192fe6de5.png)

**Comment by theonemule** [comment]
The first and second DNAT rule is on IP 172.17.0.2... is the on purpose? Is that the correct IP for the container? IP Tables uses a first-match policy, so the forwarder to 10.203.0.80 never gets hit....

**Comment by jbarraca** [comment]
yes the IP of the container is 172.17.0.2. and the IP of the VM is 10.203.0.80. 
here is a print of the 'docker inspect <container ID>':
![image](https://user-images.githubusercontent.com/33351821/36793189-a0b33176-1c94-11e8-8610-e68375fab072.png)

**Comment by theonemule** [comment]
Can you post your configuration file for Squid?

**Comment by jbarraca** [comment]
squid.conf :

> acl CONNECT method CONNECT
> 
> http_access allow manager localhost
> http_access deny manager
> 
> http_access deny to_localhost
> icp_access deny all
> htcp_access deny all
> 
> http_port 3128 transparent
> 
> hierarchy_stoplist cgi-bin ?
> access_log /var/log/squid/access.log squid
> 
> #Leave coredumps in the first cache dir
> coredump_dir /var/spool/squid
> 
> #Allow all machines to all sites
> http_access allow all
> 
> redirect_program /usr/bin/squidGuard -c /etc/squidguard/squidGuard.conf

**Comment by theonemule** [comment]
Can you give me a basic diagram of your network? Are you trying to use this machine as a FW/router, or is that separate?

Thanks!

**Comment by jbarraca** [comment]
![image](https://user-images.githubusercontent.com/33351821/36859103-4f4a9cd4-1d75-11e8-922b-27a4fda09d11.png)

It is realy just a VM and a docker container. I'm starting in this area and what i intend to do is instantiate a docker proxy to filter all HTTP/HTTPS traffic before leaving the VM. I dont know if i'm making myself clear, sorry

**Comment by theonemule** [comment]
So you want to filter all the requests leaving the VM with the Docker proxy, correct?

**Comment by jbarraca** [comment]
exactly!

**Comment by theonemule** [comment]
OK... run this command.

sysctl -w net.ipv4.conf.eth0.route_localnet=1

Also, update the rule --

iptables -t nat -A PREROUTING -i eth0 -p tcp --dport 80 -j DNAT --to **CONTAINER-IP-ADDRESS**:3128

Packets originating on a box typically don't traverse the NAT rules unless you turn this on... Try it and let me know.

**Comment by jbarraca** [comment]
well i tried it and it didnt work. so i decided to do everyting from sratch and i got this error/warning when i executed: docker build --tag docker-proxy .
![image](https://user-images.githubusercontent.com/33351821/36898743-f4b7d788-1e13-11e8-80ab-3b41d4f3f031.png)

**Comment by theonemule** [comment]
Looks like the script needs updating...

This repo is a little old. I have a working version in another project I'm working on that is based on this one, so I'll back port it.
